src/common/utility.ts

```diff
@@ -475,17 +475,18 @@ export class Utility {
         return deployment;
     }
 
-    public static serializeCreateOptions(settings: any, createOptions: any): any {
+    public static serializeCreateOptions(moduleKey: string, settings: any, createOptions: any): any {
         let optionStr: string;
         if (typeof createOptions === "string") {
-            optionStr = createOptions;
+            // remove line breaks and extra spaces
+            optionStr = JSON.stringify(JSON.parse(createOptions), null, 0);

```

Comment:
JSON parse may fail. Since now, the real createOptions will be json stringify and splitted into several chuncks (up to 8) in portal side. Like createOptions, createOptions01, createoptions02, createOptions03.....createOptions07 . So in this case if we just parse createOptions, it will fail.